lib/ProductOpener/Nutrition.pm

```diff
@@ -811,6 +815,130 @@ sub get_unit_options_for_nutrient ($nid) {
 	return \@units_options;
 }
 
+=head2 normalize_nutrient_value_string_and_modifier ( $value_ref, $modifier_ref )
+
+Each nutrient value is entered as a string (by users on the product edit form,
+or through the API). The string value may not always be numeric (e.g. it can include a < sign).
+
+This function normalizes the string value to remove signs, and stores extra information in the "modifier" field.
+
+Modifiers may also be inputted directly (e.g. through the API), and are normalized as well.
+
+=head3 Arguments
+
+=head4 string value reference $value_ref
+
+Input and output string value reference. The value will be normalized.
+
+=head4 modifier reference $modifier_ref
+
+Input and output modifier reference. The value will be normalized if it is defined.
+
+=head3 Possible return values
+
+=head4 value
+
+- 0 if the input value indicates traces
+- Number (as a string)
+- undef for 'NaN' (not a number, sometimes sent by broken API clients)
+
+=head4 modifier
+
+<, >, ≤, ≥, ~ character sign, for lesser, greater, lesser or equal, greater or equal, and about
+- (minus sign) character when the input value is - (or other dashes) : indicates that the value is not present on the package
+
+=cut
+
+# Unicode category 'Punctuation, Dash', SWUNG DASH and MINUS SIGN
+my $dashes = qr/(?:\p{Pd}|\N{U+2053}|\N{U+2212})/i;
+
+sub normalize_nutrient_value_string_and_modifier ($value_ref, $modifier_ref) {
+
+	# Note: in regexps below, the first matching alternative is used, so the longest strings must be first
+	# e.g. maximum before max
+	my $inferior_regexp = "<|&lt;|inférieur|inferieur|inferior|inf|less than|less|menor|menos";
+	my $superior_regexp = ">|&gt;|greater|more than|more|superior|mayor|más";
+	my $approximately_regexp = "~|≈|about|environ|env|aprox|alrededor";
+	my $maximum_regexp = "<=|&lt;=|\N{U+2264}|maximum|maxi|max";
+	my $minimum_regexp = ">=|&gt;=|\N{U+2265}|minimum|mini|min";
+
+	# Normalize the modifier if defined
+	if (defined ${$modifier_ref}) {
```

Comment:
good idea, it makes it more readable ! Separation of concern is a great practice :-)